api/v1alpha1/ingressreplica_types.go

```diff
@@ -0,0 +1,52 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&IngressReplica{}, &IngressReplicaList{})
+}
+
+// IngressReplicaList contains a list of [IngressReplica]
+// +kubebuilder:object:root=true
+type IngressReplicaList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IngressReplica `json:"items"`
+}
+
+// IngressReplica is the Schema for the console ingress replica
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:subresource:status
+type IngressReplica struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec of the IngressReplica
+	// +kubebuilder:validation:Required
+	Spec IngressReplicaSpec `json:"spec"`
+
+	// Status of the IngressReplica
+	// +kubebuilder:validation:Optional
+	Status Status `json:"status,omitempty"`
+}
+
+type IngressReplicaSpec struct {
```

Comment:
We should also support overriding ingress class here

Reply:
done